routes/module.js

```diff
@@ -0,0 +1,146 @@
+
+    const REQUEST_PROTOCOL = process.env.ATLAS_LINK_PROTOCOL || 'https';
+
+    router.get('/module', function(req, res, next) {
+        res.redirect('/?module=true');
+    });
+
+    /* Get marker related to module */
+    router.post('/module', function (req, res, next) {
+
+        res.header('Access-Control-Allow-Origin', '*');
+
+        var module_id=req.body.module_id;
+        if(!utils.isUUID(module_id)) {
+            return res.send(400);
+        }
+        req.session.module_id=module_id;
+
+        utils.hashToken(module_id).then(function(token) {
+            connection.query('SELECT * FROM auth WHERE token=?', [token], function (error, rows, field) {
+                if(!!error){
+                    console.log(error);
+                } else if(rows && rows.length > 0) {
+                    connection.query(show_counts_query + ' WHERE id=?', [REQUEST_PROTOCOL, req.headers.host, rows[0].atlas_id], function (error, rows, field) {
+                        if(!!error){
+                            console.log(error);
+                        } else if(rows && rows.length > 0) {
+                            res.setHeader('Content-Type', 'application/json');
+                            delete rows[0].token;
+                            res.json(rows[0]);           
+                        } else {
+                            res.send(404);
+                        }
+                    });
+                } else {
+                    res.send(404);
+                }
+            });
+        });
+    });
+
+    /* Add marker update rights for module */
+    router.post('/module/auth', function (req, res, next) {
+
+        var module_id=req.session.module_id;
+        var atlas_id=req.body.atlas_id;
+        console.log(module_id + " " + atlas_id);
+        if(!utils.isUUID(module_id) || !utils.isUUID(atlas_id)) {
+            return res.send(400);
+        }
+
+        utils.hashToken(module_id).then(function(token) {
+            connection.query('DELETE FROM auth WHERE token=?', [token], function (error, rows, field) {
+                if(!!error){
+                    console.log(error);
+                } else {
+                    connection.query('INSERT INTO auth(atlas_id,principal,token,privileges) VALUES(?,?,?,?)', [atlas_id,"module",token,"UPDATE"], function (error, rows, field) {
+                        if(!!error){
+                            console.log(error);
+                        } else {
+                            res.setHeader('Content-Type', 'application/json');
+                            res.json({ id: rows.insertId, atlas_id: atlas_id, principal: "module", privileges: "UPDATE", expires: null });           
+                        }
+                    });                            
+                }
+            });
+        });
+    });
+
+    router.post(['/module/ping', '/module/ping.php'], function(req, res, next) {
```

Comment:
Using POST and not PATCH as PATCH doesn't seem to be supported on the Java side of things.